Extension/src/main.ts

```diff
@@ -160,4 +160,20 @@ export function UpdateInsidersAccess(): void {
             migratedInsiders.Value = false;
         }
     }
+
+    // Mitigate an issue with VS Code not recognizing a programmatically installed VSIX as Prerelease.
+    // If using VS Code Insiders, and updateChannel is not explicitly set, default to Prerelease.
+    // Only do this once. If the user manually switches to Release, we don't want to switch them back to Prerelease again.
+    if (util.isVsCodeInsiders()) {
+        const insidersMitigationDone: PersistentState<boolean> = new PersistentState<boolean>("CPP.insidersMitigationDone", false);
+        if (!insidersMitigationDone.Value) {
+            if (vscode.workspace.getConfiguration("extensions", null).get<boolean>("autoUpdate")) {
+                const settings: CppSettings = new CppSettings();
+                if (settings.getWithUndefinedDefault<string>("updateChannel") === undefined) {
```

Comment:
I'm not understanding why this needs to be checked.

Reply:
If `updateChannel` was explicitly set to `Insiders` it should already have been handled by the prior check in this function.  If `updateChannel` was explicitly set to something other than `Insiders`, we should not subscribe them to prereleases.  So, this check is only needed if `updateChannel` is not explicitly set, and they are running on a VS Code Insiders build, to apply the (now missing) `Insiders` default value for `updateChannel`.  The issue this is mitigating only occurs when the user had downloaded the build via the `updateChannel` mechanism using VS Code Insiders.

Reply:
But users who explicitly set the updateChannel to "Insiders" could still be using VS Code Insiders and they won't be fixed by this?

Reply:
It sounds like you're saying the fix for https://github.com/microsoft/vscode/issues/139528 would fix the explicitly set user case.